core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto

```diff
@@ -390,3 +390,38 @@ message SQLExecutionUIData {
   repeated int64 stages = 11;
   map<int64, string> metric_values = 12;
 }
+
+message RDDOperationEdge {
+  int32 from_id = 1;
+  int32 to_id = 2;
+}
+
+message RDDOperationNode {
+  int32 id = 1;
+  string name = 2;
+  bool cached = 3;
+  bool barrier = 4;
+  string callsite = 5;
+  RDDOperationGraphWrapper.DeterministicLevel output_deterministic_level = 6;
+}
+
+message RDDOperationClusterWrapper {
+  string id = 1;
+  string name = 2;
+  repeated RDDOperationNode child_nodes = 3;
+  repeated RDDOperationClusterWrapper child_clusters = 4;
+}
+
+message RDDOperationGraphWrapper {
+  int64 stage_id = 1;
+  repeated RDDOperationEdge edges = 2;
+  repeated RDDOperationEdge outgoing_edges = 3;
+  repeated RDDOperationEdge incoming_edges = 4;
+  RDDOperationClusterWrapper root_cluster = 5;
+  enum DeterministicLevel {
```

Comment:
if `DeterministicLevel` only used by `RDDOperationNode`, should we move it into `RDDOperationNode`?  Will `DeterministicLevel` be used elsewhere in the future?

Reply:
+1, make sense

Reply:
@LuciferYang @gengliangwang done 👍🏼